src/SourceBuild/content/eng/tools/BinaryToolKit/BinaryToolKit.csproj

```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk" InitialTargets="CheckRuntimeVersion">
+
+  <PropertyGroup>
+    <TargetFramework>$(NetCurrent)</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+    <RuntimeVersion></RuntimeVersion>
+    <OutputType>Exe</OutputType>
+    <NoWarn>$(NoWarn);NU1603</NoWarn>
+  </PropertyGroup>
+
+  <Target Name="CheckRuntimeVersion">
+    <Error Condition="'$(RuntimeVersion)' == ''" Text="RuntimeVersion is not set. Please consider passing as an MSBuild Property" />
+  </Target>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.FileSystemGlobbing" Version="$(RuntimeVersion)" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="$(RuntimeVersion)" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="$(RuntimeVersion)" />
+    <PackageReference Include="System.CommandLine" Version="$(SystemCommandLineVersion)" />
```

Comment:
This needs to be treated the same way as `RuntimeVersion`. It needs to be dynamic and not defined in Versions.props otherwise it's not going to be able to be restored. The question is how do you determine what the version is? For the runtime version, it was just a matter of getting it from `dotnet --list-runtimes`. That's obviously not going to work for the System.CommandLine version. So you'd have to grab it from the package file name in the previously source built location.

Reply:
Can we just import the PackageVersions.props from the previous source-built artifacts and then use the properties it defines?

Reply:
When I testing this, I defined a version that was not in PSB (the default configured feed) and got NU1603. I resolved this by adding `<NoWarn>$(NoWarn);NU1603</NoWarn>` to the .csproj file. I assumed that this would work so long as a version of System.CommandLine is in the artifacts feed?

Reply:
Ah, great. That sounds like it should work then. Can you add a comment for the `NoWarn` that explains this?

Reply:
> Can we just import the PackageVersions.props from the previous source-built artifacts and then use the properties it defines?

Actually, that suggestion does sound like a pretty clean implementation.

Reply:
Will all package streams have a PackageVersions.props? Thinking of the case where a distro maintainer provides their own packages.

Reply:
Yes, we depend on it in our builds.